pyodide-build/pyodide_build/pypabuild.py

```diff
@@ -58,6 +58,11 @@ def replace_env(build_env: Mapping[str, str]):
 
 def install_reqs(env: IsolatedEnv, reqs: set[str]):
     env.install(remove_unisolated_requirements(reqs))
+    # Some packages (numcodecs) don't declare cython as a build dependency and
+    # only recythonize if it is present. We need them to always recythonize so
+    # we always install cython. If the reqs included some cython version already
+    # then this won't do anything.
+    env.install(["cython"])
```

Comment:
OK, but this is also a temporary solution. The right solution is to have build dependencies in meta.yaml which I need to finish one day. Installing cython for each package build "just in case" is not optimal in the long term.

Maybe it would be worth adding a TODO: note in the comments.

Reply:
Coming from conda-forge I am obviously biased here but I agree. If we could specify build/host dependencies it will be better.

Reply:
I was pretty close in https://github.com/pyodide/pyodide/pull/2132 to getting it to work (btw if you have any feedback it would be very welcome), just need to figure out the remaining CI issues. Will try to get back to it in the near future. Meanwhile, this sounds like a reasonable temporary solution.

Reply:
This all makes sense. There's a lot of work to be done on our build system, but it's improving pretty rapidly. We would really appreciate your help though. I am personally not experienced with build systems so I have to make stuff up as I go a lot, though our other maintainers can tell me when I go too far astray =)

Reply:
Absolutely would be happy to chat. Will send you an email later.